src/VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js

```diff
@@ -1,200 +1,227 @@
+                            }
+
+                            blade.parentBlade.refresh();
+                        },
+                        function(error) {
+                            bladeNavigationService.setError('Error ' + error.status, $scope.blade);
+                            blade.isLoading = false;
+                        });
                 };
-                bladeNavigationService.showBlade(newBlade, blade);
-            },
-            canExecuteMethod: function () { return true; }
-        }
-    );
-
-    function getBlogBlobName() {
-        return blade.currentEntity.name + '/' + blade.currentEntity.name + '.md'
-    }
-    function isDirty() {
-        return !angular.equals(blade.currentEntity, blade.origEntity) && blade.hasUpdatePermission();
-    }
-
-    function canSave() {
-        return isDirty() && formScope && formScope.$valid;
-    }
-
-    blade.onClose = function (closeCallback) {
-        bladeNavigationService.showConfirmationIfNeeded(isDirty(), canSave(), blade, $scope.saveChanges, closeCallback, "content.dialogs.blog-save.title", "content.dialogs.blog-save.message");
-    };
-
-    var formScope;
-    $scope.setForm = function (form) { $scope.formScope = formScope = form; };
-
-    $scope.editDictionary = function (property) {
-        var newBlade = {
-            id: "propertyDictionary",
-            isApiSave: true,
-            currentEntity: property,
-            controller: 'platformWebApp.propertyDictionaryController',
-            template: '$(Platform)/Scripts/app/dynamicProperties/blades/property-dictionary.tpl.html',
-            onChangesConfirmedFn: function () {
-                dictionaryItemsApi.query({ id: property.objectType, propertyId: property.id }, getDictionaryValuesCallback);
-            }
-        };
-        bladeNavigationService.showBlade(newBlade, blade);
-    };
 
-    var getDictionaryValuesCallback;
-    $scope.getDictionaryValues = function (property, callback) {
-        getDictionaryValuesCallback = callback;
-        dictionaryItemsApi.query({ id: property.objectType, propertyId: property.id }, callback);
-    }
+                $scope.saveChanges = async function() {
+                    blade.isLoading = true;
+
+                    const parentUrl = blade.currentEntity.url;
+                    const folderName = blade.currentEntity.name;
+                    if (!parentUrl) {
+                        contentApi.createFolder(
+                            { contentType: blade.contentType, storeId: blade.storeId },
+                            { name: folderName, parentUrl: parentUrl },
+                            $scope.saveWithMetadata
+                        );
+                    } else {
+                        if (blade.origEntity.name !== blade.currentEntity.name) {
+                            let urlsToDelete = [];
+                            let oldFilename = blade.origEntity.name + '.md';
```

Comment:
Is it ok have hardcoded file extension? We always having only "md" files?

Reply:
moved to constant